factory/config.go

```diff
@@ -61,10 +68,18 @@ func (c *Config) Validate() (err error) {
 		return fmt.Errorf("Configuration field missing")
 	}
 
+	if c.Configuration.Server == nil {
```

Comment:
This can be optional... If not present then it means we don't run http server. Remember new binary with old helm chart will fail here. We should always add code which is backward compatible with old/current helm charts.

Reply:
I agree